src/test/uint256_tests.cpp

```diff
@@ -129,140 +128,140 @@ BOOST_AUTO_TEST_CASE( comparison ) // <= >= < >
+    BOOST_CHECK_LT(R1L, MaxL);
+    BOOST_CHECK_LT(R2L, MaxL);
 
     uint160 LastS;
     for (int i = 159; i >= 0; --i) {
         uint160 TmpS;
         *(TmpS.begin() + (i>>3)) |= 1<<(7-(i&7));
-        BOOST_CHECK( LastS < TmpS );
+        BOOST_CHECK_LT(LastS, TmpS);
         LastS = TmpS;
     }
-    BOOST_CHECK( ZeroS < R1S );
-    BOOST_CHECK( R2S < R1S );
-    BOOST_CHECK( ZeroS < OneS );
-    BOOST_CHECK( OneS < MaxS );
-    BOOST_CHECK( R1S < MaxS );
-    BOOST_CHECK( R2S < MaxS );
+    BOOST_CHECK_LT(ZeroS, R1S);
+    BOOST_CHECK_LT(R2S, R1S);
+    BOOST_CHECK_LT(ZeroS, OneS);
+    BOOST_CHECK_LT(OneS, MaxS);
+    BOOST_CHECK_LT(R1S, MaxS);
+    BOOST_CHECK_LT(R2S, MaxS);
 }
 
 BOOST_AUTO_TEST_CASE( methods ) // GetHex SetHex begin() end() size() GetLow64 GetSerializeSize, Serialize, Unserialize
 {
-    BOOST_CHECK(R1L.GetHex() == R1L.ToString());
-    BOOST_CHECK(R2L.GetHex() == R2L.ToString());
-    BOOST_CHECK(OneL.GetHex() == OneL.ToString());
-    BOOST_CHECK(MaxL.GetHex() == MaxL.ToString());
+    BOOST_CHECK_EQUAL(R1L.GetHex(), R1L.ToString());
+    BOOST_CHECK_EQUAL(R2L.GetHex(), R2L.ToString());
+    BOOST_CHECK_EQUAL(OneL.GetHex(), OneL.ToString());
+    BOOST_CHECK_EQUAL(MaxL.GetHex(), MaxL.ToString());
     uint256 TmpL(R1L);
-    BOOST_CHECK(TmpL == R1L);
-    TmpL.SetHex(R2L.ToString());   BOOST_CHECK(TmpL == R2L);
-    TmpL.SetHex(ZeroL.ToString()); BOOST_CHECK(TmpL == uint256());
+    BOOST_CHECK_EQUAL(TmpL, R1L);
+    TmpL.SetHex(R2L.ToString());   BOOST_CHECK_EQUAL(TmpL, R2L);
+    TmpL.SetHex(ZeroL.ToString()); BOOST_CHECK_EQUAL(TmpL, uint256());
 
     TmpL.SetHex(R1L.ToString());
-    BOOST_CHECK(memcmp(R1L.begin(), R1Array, 32)==0);
-    BOOST_CHECK(memcmp(TmpL.begin(), R1Array, 32)==0);
-    BOOST_CHECK(memcmp(R2L.begin(), R2Array, 32)==0);
-    BOOST_CHECK(memcmp(ZeroL.begin(), ZeroArray, 32)==0);
-    BOOST_CHECK(memcmp(OneL.begin(), OneArray, 32)==0);
-    BOOST_CHECK(R1L.size() == sizeof(R1L));
-    BOOST_CHECK(sizeof(R1L) == 32);
-    BOOST_CHECK(R1L.size() == 32);
-    BOOST_CHECK(R2L.size() == 32);
-    BOOST_CHECK(ZeroL.size() == 32);
-    BOOST_CHECK(MaxL.size() == 32);
-    BOOST_CHECK(R1L.begin() + 32 == R1L.end());
-    BOOST_CHECK(R2L.begin() + 32 == R2L.end());
-    BOOST_CHECK(OneL.begin() + 32 == OneL.end());
-    BOOST_CHECK(MaxL.begin() + 32 == MaxL.end());
-    BOOST_CHECK(TmpL.begin() + 32 == TmpL.end());
-    BOOST_CHECK(GetSerializeSize(R1L) == 32);
-    BOOST_CHECK(GetSerializeSize(ZeroL) == 32);
+    BOOST_CHECK_EQUAL_COLLECTIONS(R1L.begin(), R1L.end(), R1Array, R1Array + R1L.size());
+    BOOST_CHECK_EQUAL_COLLECTIONS(TmpL.begin(), TmpL.end(), R1Array, R1Array + TmpL.size());
+    BOOST_CHECK_EQUAL_COLLECTIONS(R2L.begin(), R2L.end(), R2Array, R2Array + R2L.size());
+    BOOST_CHECK_EQUAL_COLLECTIONS(ZeroL.begin(), ZeroL.end(), ZeroArray, ZeroArray + ZeroL.size());
+    BOOST_CHECK_EQUAL_COLLECTIONS(OneL.begin(), OneL.end(), OneArray, OneArray + OneL.size());
```

Comment:
I originally did that for the +32 case, but wasn't working for the +20

Reply:
Agree it is bad form to use variable from the left side of the comparison in the right side.

Tested `std::end(R1Array)` but it would be off by 1 thanks to the null-terminator in the string, resulting in a size/distance-mismatch between the collections/iterators being tested.

Follow-up fixing this issue: #30526